### ISSUE TITLE
BUG - High CPU usage

### ISSUE THREAD

**Issue by chainslunatic** [standup-raven/standup-raven#275]
**Summary**
Standup Raven has high CPU time

**Steps To Reproduce**
1. Use plugin

**Expected behavior**
Lower CPU usage + time

**Observed Behavior**
The plugin is having up to 20% CPU load while idling and therefore is the most active process on the whole server (more than Mattermost or GitLab). The server is up for two weeks now and the plugin has almost two full days of cpu time.

**Screenshots**
<img width="1601" alt="image" src="https://user-images.githubusercontent.com/80685737/129195886-16176548-2de1-45a1-b84b-97d642a13fa7.png">

**Environment**
-  Mattermost version: 5.36.1
-  OS: Ubuntu 20.04
-  Platform: on the server itself
-  Browser (if applicable): –
-  Platform Version: –

**Comment by phpony** [comment]
Yep. Had to remove it. After 4 months of work it spikes at 100% cpu usage and eats out all memory so the server crashes. And we have only 10 people in our Matterost.

**Comment by niksubramanian** [comment]
A little message to confirm that this is also happening on our end - thanks @chainslunatic for noticing this.

@harshilsharma63 - is there any direction you could give to help figure out what could cause this? Happy to help in any way I can.

**Comment by phpony** [comment]
In my experience you can safely run `killall plugin.exe` and in a few seconds it'll safely respawn back, now without a huge CPU load. A temporarily workaround could be adding a cronjob to start once per night outside work hours:
```
15 3  * * *   root   /usr/bin/killall plugin.exe
``
It seems like this issue is in relation with posting reports to channels. Looks like each report posted adds a bit of constant CPU load into plugin process what in a matter of time sums up into a 100% load.

**Comment by niksubramanian** [comment]
I can confirm that @phpony's solution of running `killall plugin.exe` also solves the issue of CPU load on our end, but it causes the button to fill in your daily to disappear. To re-enable the button, an administrator needs to disable and re-enable the plugin in the admin console (unless you've found another workaround for this @phpony ?)

**Comment by phpony** [comment]
> but it causes the button to fill in your daily to disappear

Just Ctrl+R the page a few seconds after plugin respawn and it appears back. And if Mattermost tab was already open at time - it does not disappear. Just a caching things. Scheduling for night hours eliminates users negative feedback.

**Comment by niksubramanian** [comment]
Ok - I confirm that the button does re-appear after roughly 20-30 seconds. Thanks for the tip @phpony - I'll implement it the way you propose for now on our instance.

**Comment by chainslunatic** [comment]
@harshilsharma63 please investigate this issue.

**Comment by bbodenmiller** [comment]
Seeing high CPU usage on 3.2.2 as well.

**Comment by samzorio** [comment]
Same issue here on 3.2.2 (which seems to be wrongly labeled as 3.2.3 in github releases.

**Comment by phpony** [comment]
To make it worse latest Mattermost do not monitor and restart plugin processes, so after manually killing it, now you need to re-enable plugin in server config. Guess this project is falling out now?
